// @shopware-pwa/helpers/dist/helpers.cjs.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};

// packages/helpers/src/index.ts
__export(exports, {
  UiCategoryFilterType: () => UiCategoryFilterType,
  debounce: () => debounce,
  exportUrlQuery: () => exportUrlQuery,
  getCategoryAvailableFilters: () => getCategoryAvailableFilters,
  getCategoryAvailableSorting: () => getCategoryAvailableSorting,
  getCategoryUrl: () => getCategoryUrl,
  getCmsLayoutConfiguration: () => getCmsLayoutConfiguration,
  getCmsLink: () => getCmsLink,
  getCmsLinkTarget: () => getCmsLinkTarget,
  getCmsSections: () => getCmsSections,
  getCmsTechnicalPath: () => getCmsTechnicalPath,
  getFilterSearchCriteria: () => getFilterSearchCriteria,
  getListingAvailableFilters: () => getListingAvailableFilters,
  getListingFilters: () => getListingFilters,
  getMessagesFromErrorsArray: () => getMessagesFromErrorsArray,
  getOrderPaymentMethodId: () => getOrderPaymentMethodId,
  getOrderShippingMethodId: () => getOrderShippingMethodId,
  getProductCalculatedListingPrice: () => getProductCalculatedListingPrice,
  getProductCalculatedPrice: () => getProductCalculatedPrice,
  getProductFreeShipping: () => getProductFreeShipping,
  getProductMainImageUrl: () => getProductMainImageUrl,
  getProductMediaGallery: () => getProductMediaGallery,
  getProductName: () => getProductName,
  getProductNumber: () => getProductNumber,
  getProductOption: () => getProductOption,
  getProductOptions: () => getProductOptions,
  getProductOptionsUrl: () => getProductOptionsUrl,
  getProductPriceDiscount: () => getProductPriceDiscount,
  getProductPriceDiscountPercentage: () => getProductPriceDiscountPercentage,
  getProductProperties: () => getProductProperties,
  getProductRatingAverage: () => getProductRatingAverage,
  getProductRegularPrice: () => getProductRegularPrice,
  getProductReviews: () => getProductReviews,
  getProductSpecialPrice: () => getProductSpecialPrice,
  getProductThumbnailUrl: () => getProductThumbnailUrl,
  getProductTierPrices: () => getProductTierPrices,
  getProductUrl: () => getProductUrl,
  getSortingSearchCriteria: () => getSortingSearchCriteria,
  getStoreNavigationRoutes: () => getStoreNavigationRoutes,
  getVariantOptionsLabel: () => getVariantOptionsLabel,
  isLinkCategory: () => isLinkCategory,
  isOptionAvailableForSelectedOptions: () => isOptionAvailableForSelectedOptions,
  isProductSimple: () => isProductSimple,
  loadScript: () => loadScript,
  mapCountries: () => mapCountries,
  mapSalutations: () => mapSalutations,
  parseUrlQuery: () => parseUrlQuery
});

// packages/helpers/src/product/getProductFreeShipping.ts
function getProductFreeShipping(product) {
  return (product == null ? void 0 : product.shippingFree) || false;
}

// packages/helpers/src/product/getProductMainImageUrl.ts
function getProductMainImageUrl(product) {
  var _a, _b, _c;
  return ((_b = (_a = product == null ? void 0 : product.cover) == null ? void 0 : _a.media) == null ? void 0 : _b.url) || ((_c = product == null ? void 0 : product.cover) == null ? void 0 : _c.url) || "";
}

// packages/helpers/src/product/getProductMediaGallery.ts
function getProductMediaGallery({
  product
} = {}) {
  return product && product.media ? product.media.map((media) => {
    const smallThumb = media.media && media.media.thumbnails && media.media.thumbnails.find((thumb) => thumb.width == "400");
    const normalThumb = media.media && media.media.thumbnails && media.media.thumbnails.find((thumb) => thumb.width == "800");
    const bigThumb = media.media && media.media.thumbnails && media.media.thumbnails.find((thumb) => thumb.width == "1920");
    return {
      icon: { url: smallThumb ? smallThumb.url : media.media.url },
      mobile: { url: normalThumb ? normalThumb.url : media.media.url },
      desktop: { url: bigThumb ? bigThumb.url : media.media.url },
      alt: product.name || " "
    };
  }) : [];
}

// packages/helpers/src/product/getProductName.ts
function getProductName({ product } = {}) {
  var _a;
  if (!product) {
    return null;
  }
  return ((_a = product.translated) == null ? void 0 : _a.name) || product.name;
}

// packages/helpers/src/product/getProductNumber.ts
function getProductNumber(product) {
  return product == null ? void 0 : product.productNumber;
}

// packages/helpers/src/product/getProductOption.ts
function getProductOption({
  product,
  attribute
} = {}) {
  var _a;
  return (_a = product == null ? void 0 : product.options) == null ? void 0 : _a.find((option) => option.group && option.group.translated && option.group.translated.name === attribute);
}

// packages/helpers/src/product/getProductOptions.ts
function getProductOptions({
  product
} = {}) {
  var _a;
  return ((_a = product == null ? void 0 : product.options) == null ? void 0 : _a.map((option) => {
    var _a2, _b;
    return {
      label: ((_a2 = option.translated) == null ? void 0 : _a2.name) || option.name,
      code: option.id,
      value: ((_b = option.translated) == null ? void 0 : _b.name) || option.name,
      color: option.colorHexCode
    };
  })) || [];
}

// packages/helpers/src/product/getProductUrl.ts
function getProductUrl(product) {
  var _a, _b;
  if (!product)
    return "/";
  const seoUrl = (_b = (_a = product.seoUrls) == null ? void 0 : _a[0]) == null ? void 0 : _b.seoPathInfo;
  return seoUrl ? `/${seoUrl}` : `/detail/${product.id}`;
}

// packages/helpers/src/product/getProductOptionsUrl.ts
function getProductOptionsUrl({
  product,
  options
} = {}) {
  if (!product)
    return "";
  const variant = options && product.children && product.children.filter((variant2) => variant2.optionIds && variant2.optionIds.every((optionId) => options.includes(optionId))).shift();
  return variant && getProductUrl(variant);
}

// packages/helpers/src/product/getProductProperties.ts
function getProductProperties({
  product
} = {}) {
  var _a;
  const propertyList = (_a = product == null ? void 0 : product.properties) == null ? void 0 : _a.map((property) => {
    var _a2, _b, _c;
    return {
      name: ((_b = (_a2 = property.group) == null ? void 0 : _a2.translated) == null ? void 0 : _b.name) || "",
      value: ((_c = property.translated) == null ? void 0 : _c.name) || property.name
    };
  });
  return propertyList || [];
}

// packages/helpers/src/product/getProductRegularPrice.ts
function getProductRegularPrice(product) {
  var _a;
  return (_a = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _a.unitPrice;
}

// packages/helpers/src/product/getProductReviews.ts
function getProductReviews({
  product
} = {}) {
  if (!product || !product.productReviews) {
    return [];
  }
  return product.productReviews.map(({ id, externalUser, customerId, createdAt, content, points }) => ({
    id,
    author: externalUser ? externalUser : customerId,
    date: createdAt,
    message: content,
    rating: points
  }));
}

// packages/helpers/src/product/getProductSpecialPrice.ts
function getProductSpecialPrice(product) {
  var _a, _b;
  return (_b = (_a = product == null ? void 0 : product.calculatedPrices) == null ? void 0 : _a[0]) == null ? void 0 : _b.unitPrice;
}

// packages/helpers/src/product/getProductTierPrices.ts
function getProductTierPrices(product) {
  var _a;
  if (!product || !((_a = product.calculatedPrices) == null ? void 0 : _a.length)) {
    return [];
  }
  const size = product.calculatedPrices.length;
  return product.calculatedPrices.map(({ unitPrice, quantity }, index) => ({
    label: index === size - 1 ? `from ${quantity}` : `to ${quantity}`,
    quantity,
    unitPrice
  }));
}

// packages/helpers/src/product/getProductRatingAverage.ts
function getProductRatingAverage(product) {
  return product == null ? void 0 : product.ratingAverage;
}

// packages/helpers/src/product/getVariantOptionsLabel.ts
function getVariantOptionsLabel({
  product
} = {}) {
  if (!product || !product.options || product.options && !product.options.length) {
    return null;
  }
  let variantLabel = "";
  for (let { name } of product.options) {
    variantLabel += `${name} `;
  }
  return variantLabel.trim();
}

// packages/helpers/src/product/isProductSimple.ts
function isProductSimple({
  product
} = {}) {
  return !!(product == null ? void 0 : product.parentId) && product.id !== product.parentId;
}

// packages/helpers/src/product/getProductThumbnailUrl.ts
function getProductThumbnailUrl(product) {
  var _a, _b, _c, _d, _e;
  const coverImageUrlFallback = ((_b = (_a = product == null ? void 0 : product.cover) == null ? void 0 : _a.media) == null ? void 0 : _b.url) || "";
  const thumbnailImage = ((_e = (_d = (_c = product == null ? void 0 : product.cover) == null ? void 0 : _c.media) == null ? void 0 : _d.thumbnails) == null ? void 0 : _e.length) && product.cover.media.thumbnails.reduce(function(res, thumb) {
    return thumb.width < res.width ? thumb : res;
  }) || null;
  return (thumbnailImage == null ? void 0 : thumbnailImage.url) || coverImageUrlFallback;
}

// packages/helpers/src/product/getProductPriceDiscount.ts
function getProductPriceDiscount(product) {
  var _a, _b;
  return (_b = (_a = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _a.listPrice) == null ? void 0 : _b.discount;
}

// packages/helpers/src/product/getProductPriceDiscountPercentage.ts
function getProductPriceDiscountPercentage(product) {
  var _a, _b;
  return (_b = (_a = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _a.listPrice) == null ? void 0 : _b.percentage;
}

// packages/helpers/src/product/getProductCalculatedPrice.ts
function getProductCalculatedPrice(product) {
  var _a;
  return (_a = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _a.unitPrice;
}

// packages/helpers/src/product/getProductCalculatedListingPrice.ts
function getProductCalculatedListingPrice(product) {
  var _a, _b, _c;
  return ((_b = (_a = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _a.listPrice) == null ? void 0 : _b.price) || ((_c = product == null ? void 0 : product.calculatedPrice) == null ? void 0 : _c.unitPrice);
}

// packages/helpers/src/product/isOptionAvailableForSelectedOptions.ts
function isOptionAvailableForSelectedOptions(currentAttribute, selectedOptionId, { code: optionId } = {}, allOptions, allSelectedOptions) {
  if (!currentAttribute || !selectedOptionId || !optionId || !allOptions || !allSelectedOptions) {
    return false;
  }
  if (optionId == selectedOptionId) {
    return true;
  }
  let matchingOptions = false;
  Object.entries(allSelectedOptions).forEach(([attribute, selectedId]) => {
    if (attribute !== currentAttribute) {
      allOptions[attribute].forEach((otherOption) => {
        if (otherOption.code === selectedId) {
          matchingOptions = otherOption["matchingIds"].includes(optionId);
        }
      });
    }
  });
  return matchingOptions;
}

// packages/helpers/src/ui-interfaces.ts
var UiCategoryFilterType;
(function(UiCategoryFilterType2) {
  UiCategoryFilterType2["range"] = "range";
  UiCategoryFilterType2["term"] = "term";
  UiCategoryFilterType2["max"] = "max";
  UiCategoryFilterType2["entity"] = "entity";
})(UiCategoryFilterType || (UiCategoryFilterType = {}));

// packages/helpers/src/category/getCategoryAvailableFilters.ts
var convertTermFilterValues = (values) => {
  return values.map(({ key, count }) => ({
    value: key,
    label: key,
    count
  }));
};
var convertEntityFilterValues = (values) => {
  return !values ? [] : Object.entries(values).map(([valueId, { name, colorHexCode }]) => {
    let filterValue = {
      value: valueId,
      label: name
    };
    if (colorHexCode) {
      filterValue = Object.assign({}, filterValue, { color: colorHexCode });
    }
    return filterValue;
  });
};
var convertOptionsByType = ({
  type,
  values
}) => {
  switch (type) {
    case UiCategoryFilterType.term:
      return convertTermFilterValues(values);
    case UiCategoryFilterType.entity:
      return convertEntityFilterValues(values);
    default:
      return values;
  }
};
function getCategoryAvailableFilters({
  filters
} = {}) {
  if (!filters) {
    return [];
  }
  const filtersTransformed = Object.entries(filters).map(([filterCode, { name, values, type }]) => {
    return {
      name: name || filterCode,
      type,
      options: convertOptionsByType({
        type,
        values
      })
    };
  });
  return filtersTransformed;
}

// packages/helpers/src/category/getCategoryAvailableSorting.ts
function getCategoryAvailableSorting({
  sorting
} = {}) {
  if (!sorting) {
    return [];
  }
  const sortingTransformed = Object.entries(sorting).map(([sortingCode, { active }]) => ({
    name: sortingCode,
    active,
    field: sortingCode.split("-")[0],
    order: sortingCode.split("-")[1]
  }));
  return sortingTransformed;
}

// packages/helpers/src/category/getCategoryUrl.ts
var getCategoryUrl = (category) => {
  var _a, _b, _c;
  if (!category)
    return "/";
  switch (category.type) {
    case "link":
      return ((_a = category.translated) == null ? void 0 : _a.externalLink) || category.externalLink || "/";
    case "folder":
      return "/";
    default:
      return ((_c = (_b = category.seoUrls) == null ? void 0 : _b[0]) == null ? void 0 : _c.seoPathInfo) ? `/${category.seoUrls[0].seoPathInfo}` : category.id ? `/navigation/${category.id}` : "/";
  }
};
var isLinkCategory = (category) => (category == null ? void 0 : category.type) === "link";

// packages/helpers/src/country/mapCountries.ts
function mapCountries(countries) {
  if (!(countries == null ? void 0 : countries.length))
    return [];
  const mappedCountries = countries.map((country) => {
    var _a;
    return {
      name: ((_a = country.translated) == null ? void 0 : _a.name) || country.name,
      id: country.id
    };
  });
  return mappedCountries.filter((country) => country.name !== null);
}

// packages/helpers/src/cms/getCmsLink.ts
function getCmsLink(content) {
  var _a;
  return ((_a = content == null ? void 0 : content.data) == null ? void 0 : _a.url) || "";
}

// packages/helpers/src/cms/getCmsLinkTarget.ts
function getCmsLinkTarget(content) {
  var _a;
  const inNewTab = (_a = content == null ? void 0 : content.data) == null ? void 0 : _a.newTab;
  return inNewTab ? "_blank" : "_self";
}

// packages/helpers/src/cms/getCmsSections.ts
function getCmsSections(content) {
  return (content == null ? void 0 : content.sections) || [];
}

// packages/commons/interfaces/models/content/cms/CmsPage.ts
var PageType;
(function(PageType2) {
  PageType2["PRODUCT_DETAIL_PAGE"] = "frontend.detail.page";
  PageType2["NAVIGATION_PAGE"] = "frontend.navigation.page";
})(PageType || (PageType = {}));
var CmsPageType;
(function(CmsPageType2) {
  CmsPageType2["DEFAULT"] = "default";
  CmsPageType2["PRODUCT_LISTING"] = "product_list";
  CmsPageType2["LANDING_PAGE"] = "landingpage";
})(CmsPageType || (CmsPageType = {}));
var CmsSlotType;
(function(CmsSlotType2) {
  CmsSlotType2["IMAGE"] = "image";
  CmsSlotType2["PRODUCT_SLIDER"] = "product-slider";
  CmsSlotType2["PRODUCT_LISTING"] = "product-listing";
  CmsSlotType2["PRODUCT_BOX"] = "product-box";
  CmsSlotType2["SLOT"] = "slot";
  CmsSlotType2["TEXT"] = "text";
})(CmsSlotType || (CmsSlotType = {}));
var SectionType;
(function(SectionType2) {
  SectionType2["DEFAULT"] = "default";
})(SectionType || (SectionType = {}));
var SizingMode;
(function(SizingMode2) {
  SizingMode2["BOXED"] = "boxed";
})(SizingMode || (SizingMode = {}));
var MobileBehavior;
(function(MobileBehavior2) {
  MobileBehavior2["BOXED"] = "boxed";
  MobileBehavior2["WRAP"] = "wrap";
})(MobileBehavior || (MobileBehavior = {}));
var BackgroundMediaMode;
(function(BackgroundMediaMode2) {
  BackgroundMediaMode2["COVER"] = "cover";
})(BackgroundMediaMode || (BackgroundMediaMode = {}));

// packages/helpers/src/cms/getCmsTechnicalPath.ts
var PRODUCT_PAGE_PREFIX = "/detail";
var NAVIGATION_PAGE_PREFIX = "/navigation";
var getProductTechnicalPath = (product) => `${PRODUCT_PAGE_PREFIX}/${product.id}`;
var getNavigationTechnicalPath = (cmsPage) => `${NAVIGATION_PAGE_PREFIX}/${cmsPage.resourceIdentifier}`;
function getCmsTechnicalPath(page) {
  if (!(page == null ? void 0 : page.resourceType)) {
    return;
  }
  switch (page.resourceType) {
    case PageType.NAVIGATION_PAGE:
      return getNavigationTechnicalPath(page);
    case PageType.PRODUCT_DETAIL_PAGE:
      return getProductTechnicalPath(page.product);
    default:
      throw Error(`Cannot extract a technical URL for provided page type: ${page.resourceType}`);
  }
}

// packages/helpers/src/cms/getCmsLayoutConfiguration.ts
function getCmsLayoutConfiguration(content) {
  if (!content) {
    return {
      cssClasses: null,
      layoutStyles: {}
    };
  }
  return {
    cssClasses: content.cssClass,
    layoutStyles: {
      backgroundColor: content.backgroundColor,
      backgroundImage: content.backgroundMedia ? `url(${content.backgroundMedia.url})` : null,
      marginBottom: content.marginBottom,
      marginLeft: content.marginLeft,
      marginRight: content.marginRight,
      marginTop: content.marginTop
    }
  };
}

// packages/helpers/src/urlHelpers.ts
var import_query_string = __toModule(require("query-string"));
function parseUrlQuery(query) {
  const searchCriteria = {};
  if (!query || typeof query !== "object") {
    return searchCriteria;
  }
  Object.keys(query).forEach((key) => {
    try {
      searchCriteria[key] = typeof query[key] === "string" && ["{", "["].includes(query[key].charAt(0)) ? JSON.parse(query[key]) : query[key];
    } catch (e) {
      console.error("[helpers][parseUrlQuery] Problem with resolving url param: " + key);
    }
  });
  return searchCriteria;
}
function exportUrlQuery(searchCriteria) {
  if (!searchCriteria) {
    return;
  }
  const sC = searchCriteria;
  const query = {};
  Object.keys(searchCriteria).forEach((key) => {
    query[key] = JSON.stringify(sC[key]);
  });
  return import_query_string.default.stringify(query);
}

// packages/commons/interfaces/search/SearchFilter.ts
var SearchFilterType;
(function(SearchFilterType2) {
  SearchFilterType2["EQUALS"] = "equals";
  SearchFilterType2["CONTAINS"] = "contains";
  SearchFilterType2["EQUALS_ANY"] = "equalsAny";
  SearchFilterType2["NOT"] = "not";
  SearchFilterType2["MULTI"] = "multi";
  SearchFilterType2["RANGE"] = "range";
  SearchFilterType2["MAX"] = "max";
})(SearchFilterType || (SearchFilterType = {}));

// packages/helpers/src/listing/filters.ts
var createMultiFilter = (operator, queries) => ({
  type: SearchFilterType.MULTI,
  operator,
  queries
});
var createRangeFilter = (filterData, field) => ({
  type: SearchFilterType.RANGE,
  field,
  parameters: filterData
});
var createMaxFilter = (filterData, field) => ({
  type: SearchFilterType.MAX,
  field,
  max: filterData.max
});
var createEqualsFilter = (value, field) => ({
  type: SearchFilterType.EQUALS,
  field,
  value
});
var createEqualsAnyFilter = (values, field = "propertyIds") => {
  if (!Array.isArray(values)) {
    throw "EqualsAnyFilter value is not an array";
  }
  const valuesFiltered = values.filter((value) => !!value);
  if (!valuesFiltered.length) {
    throw "EqualsAnyFilter has no values provided";
  }
  return {
    type: SearchFilterType.EQUALS_ANY,
    field,
    value: valuesFiltered
  };
};
var extractFilter = (filterCode, filterData) => {
  let extractedFilter = null;
  switch (filterCode) {
    case "price":
      extractedFilter = createRangeFilter(filterData, filterCode);
      break;
    case "shipping-free":
      extractedFilter = createMaxFilter(filterData, filterCode);
      break;
    case "rating":
      extractedFilter = createMaxFilter(filterData, filterCode);
      break;
    case "categoryTree":
      extractedFilter = createEqualsFilter(filterData.shift(), "product.categoriesRo.id");
      break;
    case "manufacturer":
      extractedFilter = createEqualsAnyFilter(filterData, "manufacturerId");
      break;
    default:
      const subFilters = [];
      subFilters.push(createEqualsAnyFilter(filterData));
      subFilters.push(createEqualsAnyFilter(filterData, "optionIds"));
      extractedFilter = createMultiFilter("OR", subFilters);
  }
  return extractedFilter;
};
var getFilterSearchCriteria = (selectedFilters) => {
  const filters = [];
  if (!selectedFilters) {
    return filters;
  }
  for (const filterName of Object.keys(selectedFilters)) {
    try {
      filters.push(extractFilter(filterName, selectedFilters[filterName]));
    } catch (error) {
      console.warn("helpers:getFilterSearchCriteria:extractFilter", error);
    }
  }
  return filters;
};
var getSortingSearchCriteria = (selectedSorting) => {
  var _a;
  if (!selectedSorting) {
    return {};
  }
  return {
    field: selectedSorting.field,
    name: ((_a = selectedSorting.translated) == null ? void 0 : _a.name) || selectedSorting.name,
    desc: selectedSorting.order === "desc"
  };
};

// packages/helpers/src/listing/getListingAvailableFilters.ts
var getFilterType = (aggregation) => {
  if (aggregation.entities && Array.isArray(aggregation.entities)) {
    return UiCategoryFilterType.entity;
  }
  if (aggregation.hasOwnProperty("max") && aggregation.hasOwnProperty("min")) {
    return UiCategoryFilterType.range;
  }
  if (aggregation.hasOwnProperty("max") && !aggregation.min) {
    return UiCategoryFilterType.max;
  }
  throw new Error("Unrecognized type");
};
var extractEntityTypeFilter = (name, options) => ({
  name,
  type: UiCategoryFilterType.entity,
  options: options.map((filterData) => {
    var _a;
    return __spreadValues({
      label: ((_a = filterData.translated) == null ? void 0 : _a.name) || filterData.name,
      value: filterData.id,
      color: filterData.colorHexCode
    }, filterData);
  })
});
function getListingAvailableFilters(aggregations) {
  var _a;
  if (!aggregations) {
    return [];
  }
  const transformedFilters = [];
  for (const [aggregationName, aggregation] of Object.entries(aggregations)) {
    try {
      const filterType = getFilterType(aggregation);
      if (filterType === UiCategoryFilterType.entity) {
        if (aggregationName === "properties") {
          for (const property of aggregation.entities) {
            transformedFilters.push(extractEntityTypeFilter(((_a = property.translated) == null ? void 0 : _a.name) || property.name, property.options));
          }
        } else {
          transformedFilters.push(extractEntityTypeFilter(aggregationName, aggregation.entities));
        }
      } else {
        transformedFilters.push(__spreadValues({
          name: aggregationName,
          type: filterType
        }, aggregation));
      }
    } catch (error) {
      console.warn(`[helpers][getListingAvailableFilters][getFilterType]: ${error} | ${aggregationName}`);
    }
  }
  return transformedFilters;
}

// packages/helpers/src/listing/getListingFilters.ts
var getFilter = (code, aggregation) => {
  var _a;
  return __spreadValues({
    label: ((_a = aggregation.translated) == null ? void 0 : _a.name) || code,
    code
  }, aggregation);
};
function getListingFilters(aggregations) {
  if (!aggregations) {
    return [];
  }
  const transformedFilters = [];
  for (const [aggregationName, aggregation] of Object.entries(aggregations)) {
    if (aggregationName === "properties" && aggregation.entities) {
      for (const property of aggregation.entities) {
        transformedFilters.push(getFilter(aggregationName, property));
      }
    } else if (!["properties", "options"].includes(aggregationName)) {
      transformedFilters.push(getFilter(aggregationName, aggregation));
    }
  }
  return transformedFilters;
}

// packages/helpers/src/navigation/getStoreNavigationRoutes.ts
function getStoreNavigationRoutes(navigationElements) {
  return navigationElements.map((element) => {
    var _a, _b, _c;
    return {
      routeLabel: ((_a = element.translated) == null ? void 0 : _a.name) || element.name,
      isExternal: !!element.externalLink,
      routePath: element.externalLink || ((_c = (_b = element.seoUrls) == null ? void 0 : _b[0]) == null ? void 0 : _c.seoPathInfo) && `/${element.seoUrls[0].seoPathInfo}`,
      children: element.children && getStoreNavigationRoutes(element.children)
    };
  });
}

// packages/helpers/src/salutation/mapSalutations.ts
function mapSalutations(salutations) {
  if (!(salutations == null ? void 0 : salutations.length))
    return [];
  const mappedSalutations = salutations.map((salutation) => {
    var _a;
    return {
      name: (_a = salutation.displayName) != null ? _a : salutation.salutationKey,
      id: salutation.id
    };
  });
  return mappedSalutations.filter((mappedSalutation) => mappedSalutation.name !== null);
}

// packages/helpers/src/error/index.ts
function getMessagesFromErrorsArray(errors) {
  if (!(errors == null ? void 0 : errors.length) || !Array.isArray(errors)) {
    return [];
  }
  return errors.map(({ detail, source }) => detail && source && source.pointer && `${source.pointer.substring(1)}: ${detail}`).filter((message) => message);
}

// packages/helpers/src/order/getOrderPaymentMethodId.ts
function getOrderPaymentMethodId(order) {
  var _a;
  const openTransaction = (_a = order == null ? void 0 : order.transactions) == null ? void 0 : _a.find(({ stateMachineState }) => (stateMachineState == null ? void 0 : stateMachineState.technicalName) === "open");
  return openTransaction == null ? void 0 : openTransaction.paymentMethodId;
}

// packages/helpers/src/order/getOrderShippingMethodId.ts
function getOrderShippingMethodId(order) {
  var _a;
  const openDelivery = (_a = order == null ? void 0 : order.deliveries) == null ? void 0 : _a.find(({ stateMachineState }) => (stateMachineState == null ? void 0 : stateMachineState.technicalName) === "open");
  return openDelivery == null ? void 0 : openDelivery.shippingMethodId;
}

// packages/helpers/src/plugins/loadScript.ts
function loadScript(src) {
  return new Promise(function(resolve, reject) {
    if (!document || document.querySelector('script[src="' + src + '"]')) {
      resolve();
      return;
    }
    const el = document.createElement("script");
    el.type = "text/javascript";
    el.async = true;
    el.src = src;
    el.addEventListener("load", resolve);
    el.addEventListener("error", reject);
    el.addEventListener("abort", reject);
    document.head.appendChild(el);
  });
}

// packages/helpers/src/debounce.ts
function debounce(fn, delay = 300) {
  let prevTimer = null;
  return (...args) => {
    if (prevTimer) {
      clearTimeout(prevTimer);
    }
    prevTimer = window.setTimeout(() => {
      fn(...args);
      prevTimer = null;
    }, delay);
  };
}
